flexmeasures/api/v3_0/sensors.py

```diff
@@ -1508,3 +1511,79 @@ def get_status(self, id, sensor):
         status_data = serialize_sensor_status_data(sensor=sensor)
 
         return {"sensors_data": status_data}, 200
+
+    @route("/<sensor>/forecasts/trigger", methods=["POST"])
+    @use_kwargs({"sensor": SensorIdField()}, location="path")
+    @permission_required_for_context("read", ctx_arg_name="sensor")
+    def trigger_forecast(self, sensor: Sensor):
+        """
+        .. :quickref: Forecasts; Trigger forecasting job for one sensor
+        ---
+        post:
+          summary: Trigger forecasting job for one sensor
+          description: |
+            Launch a forecasting pipeline for the given sensor asynchronously.
+            This reuses the same validation as the CLI command `flexmeasures add forecasts`.
+
+            Example:
+            ```
+            {
+              "start_date": "2026-01-01T00:00:00+01:00",
+              "start_predict_date": "2026-01-15T00:00:00+01:00",
+              "end_date": "2026-01-17T00:00:00+01:00"
+            }
+            ```
+          responses:
+            200:
+              description: Forecasting job queued
+              content:
+                application/json:
+                  example:
+                    status: "PROCESSED"
+                    forecast_jobs: ["b3d26a8a-7a43-4a9f-93e1-fc2a869ea97b"]
+                    message: "Forecasting job has been queued."
+          tags:
+            - Sensors
+        """
+
+        try:
+            # Load and validate JSON payload
+            parameters = request.get_json()
+            parameters["sensor"] = sensor.id
+            # Ensure the forecast is run as a job on a forecasting queue
+            parameters["as_job"] = True
+
+            # Set up forecaster source
+            source = parameters.pop("source", None)
+
+            # Set forecaster model
+            model = parameters.pop("model", "TrainPredictPipeline")
+
+            # Instantiate the forecaster
+            forecaster = get_data_generator(
+                source=source,
+                model=model,
+                config={},
+                save_config=True,
+                data_generator_type=Forecaster,
+            )
+
+            # Queue forecasting job
+            results = forecaster.compute(parameters=parameters)
+
+            # Extract job ID (UUID)
+            job_ids = []
+            for result in results:
+                job_ids.extend(result.values())
+
+            # Commit DB transaction
+            db.session.commit()
+
+            d, s = request_processed()
+            return dict(forecast_jobs=job_ids, **d), s
+
+        except ValidationError as e:
+            return invalid_flex_config(e.messages)
+        except Exception as e:
+            current_app.logger.exception("Forecast job failed to enqueue.")
+            return invalid_flex_config(str(e))
```

Comment:
`invalid_flex_config` looks like something specific to scheduling.

Reply:
I see, thanks for the context. I guess renaming it to `def unprocessable_entity` would make sense.